cpp/ql/src/semmle/code/cpp/ir/implementation/raw/internal/IRConstruction.qll

```diff
@@ -179,8 +171,10 @@ cached private module Cached {
 import CachedForDebugging
 cached private module CachedForDebugging {
   cached string getTempVariableUniqueId(IRTempVariable var) {
-    result = getTempVariableTranslatedElement(var).getId() + ":" +
-      getTempVariableTagId(var.getTag())
+    exists(TranslatedElement element |
+      var = element.getTempVariable(_) and
```

Comment:
Does this need to be `var = element.getTempVariable(var.getTag())`? I'm not sure the proposed change will work if the `TranslatedElement` has multiple temp variables. I'm not sure if this can occur in practice today, though.

Reply:
My intention here is to effectively replace `getTempVariableTranslatedElement(var)` with `any(TranslatedElement element | var = element.getTempVariable(_))`. Both expressions should give the (unique) element that's associated with `var`, right?

Reply:
You're right. Upon rereading, it looks fine. I don't know what I was thinking before.